src/utils/task-filtering.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Filter an array of tasks based on their type.
+ *
+ * @param {Array} tasks - The array of tasks to be filtered.
+ * @param {string} type - The type to filter by.
+ * @returns {Array} The filtered array of tasks.
+ */
+function filterTasksByType(tasks, type) {
+  return tasks.filter(task => task.type === type);
+}
+
+/**
+ * Filter an array of tasks based on their status.
+ *
+ * @param {Array} tasks - The array of tasks to be filtered.
+ * @param {string} status - The status to filter by.
+ * @returns {Array} The filtered array of tasks.
+ */
+function filterTasksByStatus(tasks, status) {
+  return tasks.filter(task => task.status === status);
+}
+
+/**
+ * Filter an array of tasks based on their creation time.
```

Comment:
I think personally the only filter we need is for oldest to newest and newest to oldest, I would like others input but I wouldn't think its necessary to filter by a specific date (like before 12/5 or after). That would also introduce a good bit of overhead to do error handling for whatever the user inputs @davidtso1219 @mottur let me know your thoughts

Reply:
i agree, newest to oldest and vice versa is probably all that's needed.

Reply:
yea this function is unlikely to be needed